extensions/netty/runtime/src/main/java/io/quarkus/netty/runtime/graal/NettySubstitutions.java

```diff
@@ -515,6 +512,50 @@ public long memoryAddress() {
 
 }
 
+@TargetClass(className = "io.netty.handler.codec.compression.Brotli")
+final class Target_io_netty_handler_codec_compression_Brotli {
+    @Substitute
+    public static void ensureAvailability() throws Throwable {
+        throw new ClassNotFoundException();
+    }
+
+    @Substitute
+    public static boolean isAvailable() {
+        return false;
+    }
+}
```

Comment:
I haven't checked the code but just throwing a random idea: would this work if you substituted just this class and deleted `io.netty.handler.codec.compression.BrotliDecoder` entirely?

Maybe we can brainstorm tomorrow morning?

Reply:
Apparently, GraalVM is not clever enough to understand that the code in `HttpContentDecompressor` won't be accessed.

So I suppose we would need to create two extensions, one for the HTTP codec and one for the HTTP2 codec ? I can do that tomorrow if we think that's the way to go.

Reply:
The problem was the call sites, I did something that deletes the `BrotliDecoder`.